49/Gealber/crawl_verifier.py

```diff
@@ -0,0 +1,187 @@
+import concurrent.futures
+import threading
+from urllib.request import Request, urlopen, URLError
+from urllib.parse import urljoin, urlparse
+from bs4 import BeautifulSoup
+import sqlite3
+import feedparser
+from datetime import date
+import queue
+import time
+import logging
+
+conn = sqlite3.connect('urls.db')
+c = conn.cursor()
+c.execute('CREATE TABLE Urls (urls VARCHAR)')
+links_to_crawl = queue.Queue(maxsize=10)
+
+
+class Crawl:
+    base_url = ''
+    error_links = set()
+    crawled_links = set()
+    headers = {}
+
+    def __init__(self, headers, base_url):
+        Crawl.headers = headers
+        Crawl.base_url = base_url
+
+    @staticmethod
+    def crawl(thread_name, url, links_to_crawl):
+        """
+        This method connect to the given url and put it in the corresponding set() or queue.
+        If there's an error when parsing the url is added to error_links set.
+        Otherwise extract the links of the page with BeautifulSoup and pass this links to the method
+        enqueue_links() to well ....be enqueued.
+        :param thread_name: Current thread name
+        :param url: url to connect
+        :param links_to_crawl: queue with the links extracted.
+        """
+        try:
```

Comment:
make this try block narrower, `requests` would provide you with a `raise_for_status` = convenient